notion-core/src/main/scala/com/ambiata/notion/core/LocationIO.scala

```diff
@@ -110,6 +108,19 @@ case class LocationIO(configuration: Configuration, s3Client: AmazonS3Client) {
       }}.as(state)
     }
 
+  /** read a file as a stream of bytes and compute some state value */
+  def streamBytes[A](location: Location, empty: => A, bufferSize: Int = 4096)(f: (ReadBytes, A) => A): RIO[A] =
+    RIO.io(empty).flatMap { s =>
+      var state = s
+      readUnsafe(location) { in => RIO.io {
+        val buffer = Array.ofDim[Byte](bufferSize)
+        var length = 0
+        while ({ length = in.read(buffer, 0, buffer.length); length != -1 }) {
+          state = f(ReadBytes(buffer, 0, length), state)
```

Comment:
I had a closer look at `Buffer` and the `array` part of `Buffer` immutable. How would that work in that case?